libwayshot/src/lib.rs

```diff
@@ -93,383 +57,373 @@ pub struct CaptureRegion {
-                frame_copy.frame_format.height,
-                frame_copy.frame_color_type,
-            )?;
-            output_file.flush()?;
-        }
-        EncodingFormat::Png => {
-            PngEncoder::new(&mut output_file).write_image(
-                &frame_copy.frame_mmap,
-                frame_copy.frame_format.width,
-                frame_copy.frame_format.height,
-                frame_copy.frame_color_type,
-            )?;
-            output_file.flush()?;
-        }
-        EncodingFormat::Qoi => {
-            QoiEncoder::new(&mut output_file).write_image(
-                &frame_copy.frame_mmap,
-                frame_copy.frame_format.width,
-                frame_copy.frame_format.height,
-                frame_copy.frame_color_type,
-            )?;
-            output_file.flush()?;
-        }
-        EncodingFormat::Ppm => {
-            let rgb8_data = if let ColorType::Rgba8 = frame_copy.frame_color_type {
-                let mut data = Vec::with_capacity(
-                    (3 * frame_copy.frame_format.width * frame_copy.frame_format.height) as _,
-                );
-                for chunk in frame_copy.frame_mmap.chunks_exact(4) {
-                    data.extend_from_slice(&chunk[..3]);
-                }
-                data
-            } else {
-                unimplemented!("Currently only ColorType::Rgba8 is supported")
-            };
-
-            PnmEncoder::new(&mut output_file)
-                .with_subtype(pnm::PnmSubtype::Pixmap(pnm::SampleEncoding::Binary))
-                .write_image(
-                    &rgb8_data,
-                    frame_copy.frame_format.width,
-                    frame_copy.frame_format.height,
-                    ColorType::Rgb8,
-                )?;
-            output_file.flush()?;
+    /// Take a screenshot from all of the specified outputs.
+    pub fn screenshot_outputs(
+        &self,
+        outputs: Vec<OutputInfo>,
+        cursor_overlay: bool,
+    ) -> Result<RgbaImage> {
+        if outputs.is_empty() {
+            bail!("no outputs supplied");
         }
+
+        let x1 = outputs
+            .iter()
+            .map(|output| output.dimensions.x)
+            .min()
+            .unwrap();
+        let y1 = outputs
+            .iter()
+            .map(|output| output.dimensions.y)
+            .min()
+            .unwrap();
+        let x2 = outputs
+            .iter()
+            .map(|output| output.dimensions.x + output.dimensions.width)
+            .max()
+            .unwrap();
+        let y2 = outputs
+            .iter()
+            .map(|output| output.dimensions.y + output.dimensions.height)
+            .max()
+            .unwrap();
+        let capture_region = CaptureRegion {
+            x_coordinate: x1,
+            y_coordinate: y1,
+            width: x2 - x1,
+            height: y2 - y1,
+        };
+        self.screenshot((Transform::Normal, capture_region), cursor_overlay)
```

Comment:
I think the transform should decided later, by output them self, so this one, Transform I think is needless, when I know a region, I will never know what transform every part is